Jenkinsfile

```
@@ -0,0 +1,191 @@
+            script{
+              archiveZap failHighAlerts: 1, failMediumAlerts: 1, failLowAlerts: 1
+            }  
+            publishHTML([allowMissing: false, alwaysLinkToLastBuild: true, keepAll: true, reportDir: '/var/lib/jenkins/iudx/rs-proxy/Newman/report/', reportFiles: 'report.html', reportTitles: '', reportName: 'Integration Test Report'])
+          }
+        }
+        failure{
+          error "Test failure. Stopping pipeline execution!"
+        }
+        cleanup{
+          script{
+            sh 'docker compose -f docker-compose.test.yml down --remove-orphans'
+          } 
+        }
+      }
+    }
+
+    stage('Continuous Deployment') {
+      when {
+        allOf {
+          anyOf {
+            changeset "docker/**"
+            changeset "docs/**"
+            changeset "pom.xml"
+            changeset "src/main/**"
+            triggeredBy cause: 'UserIdCause'
+          }
+          expression {
+            return env.GIT_BRANCH == 'origin/master';
+          }
+        }
+      }
+      stages {
+        stage('Push Images') {
+          steps {
+            script {
+              docker.withRegistry( registryUri, registryCredential ) {
+                devImage.push("5.5.0-alpha-${env.GIT_HASH}")
+                deplImage.push("5.5.0-alpha-${env.GIT_HASH}")
+              }
+            }
+          }
+        }
+        stage('Docker Swarm deployment') {
+          steps {
+            script {
+              sh "ssh azureuser@docker-swarm 'docker service update rs-proxy_rs-proxy --image ghcr.io/datakaveri/rs-proxy-depl:5.5.0-alpha-${env.GIT_HASH}'"
+              sh 'sleep 10'
+            }
+          }
+          post{
+            failure{
+              error "Failed to deploy image in Docker Swarm"
+            }
+          }          
+        }
+        stage('Integration test on swarm deployment') {
+          steps {
+            node('built-in') {
+              script{
+                sh 'newman run /var/lib/jenkins/iudx/fs/Newman/IUDX-Resource-Proxy-Server-Consumer-APIs.postman_collection_5.5.0.json -e /home/ubuntu/configs/cd/rs-proxy-postman-env.json --insecure -r htmlextra --reporter-htmlextra-export /var/lib/jenkins/iudx/rs-proxy/Newman/report/cd-report.html --reporter-htmlextra-skipSensitiveData'
+              }
+            }
+          }
+          post{
+            always{
+              node('built-in') {
+                script{
+                  publishHTML([allowMissing: false, alwaysLinkToLastBuild: true, keepAll: true, reportDir: '/var/lib/jenkins/iudx/rs-proxy/Newman/report/', reportFiles: 'cd-report.html', reportTitles: '', reportName: 'Docker-Swarm Integration Test Report'])
+                }
+              }
+            }
+            failure{
+              error "Test failure. Stopping pipeline execution!"
+            }
+          }
+        }
+      }
+    }
+  }
+  post{
+    failure{
+      script{
+        if (env.GIT_BRANCH == 'origin/master')
```

branch is not master, it's main
```suggestion
        if (env.GIT_BRANCH == 'origin/main')
```